features/step_definitions/brief_response_steps.rb

```diff
@@ -24,7 +24,7 @@
 end
 
 Given 'that supplier is on that framework' do
-  submit_supplier_declaration(@framework['slug'], @supplier["id"], {})
+  submit_supplier_declaration_and_sign_agreement(@framework['slug'], @supplier["id"], {'status': 'complete'})
```

Comment:
I think it would be better to have a new step `And that supplier has returned a signed agreement`, which can follow on from `that supplier is on that framework` in the scenario/setup steps.

If "that supplier is on that framework" also submits the agreement then we won't easily be able to build out functional tests for the signing agreement flow - i.e. to test that flow we'll need a supplier on the framework who hasn't already submitted their agreement.

And being on a framework and having returned the thing are conceptually and prctically different things.